File: domain/src/main/kotlin/com/readler/domain/repository/AiRepository.kt
package com.readler.domain.repository

interface AiRepository {
    suspend fun isEnabled(): Boolean
    suspend fun setEnabled(enabled: Boolean)
    suspend fun setApiKey(apiKey: String)
    suspend fun getApiKey(): String
    suspend fun getCapability(): AiCapability
    suspend fun askQuestion(bookId: String, bookTitle: String, author: String?, question: String): String
    suspend fun explainSelection(bookId: String, bookTitle: String, selectedText: String): String
    suspend fun translateSelection(bookId: String, bookTitle: String, selectedText: String, targetLanguage: String): String
    suspend fun suggestSimilarBooks(bookId: String, bookTitle: String, author: String?): String
    suspend fun summarizeSection(bookId: String, bookTitle: String, sectionText: String): String
    suspend fun enqueueRequest(request: AiQueuedRequest)
    suspend fun flushQueuedRequests(): Int
}

File: lib/domain/repository/ai_repository.dart
import 'ai_capability.dart';
import 'ai_queued_request.dart';

abstract class AiRepository {
  Future<bool> isEnabled();
  Future<void> setEnabled(bool enabled);
  Future<void> setApiKey(String apiKey);
  Future<String> getApiKey();
  Future<AiCapability> getCapability();
  Future<String> askQuestion(
      String bookId, String bookTitle, String? author, String question);
  Future<String> explainSelection(
      String bookId, String bookTitle, String selectedText);
  Future<String> translateSelection(String bookId, String bookTitle,
      String selectedText, String targetLanguage);
  Future<String> suggestSimilarBooks(
      String bookId, String bookTitle, String? author);
  Future<String> summarizeSection(
      String bookId, String bookTitle, String sectionText);
  Future<void> enqueueRequest(AiQueuedRequest request);
  Future<int> flushQueuedRequests();
}
